rpc-websocket-wrappers/client.js

```diff
@@ -12,24 +12,26 @@ const { TimeoutError }			= async_with_timeout;
 class WebSocket extends RPCWebSocket {
 
     opened ( timeout = 1000 ) {
-	log.silly(".opened() readyState: %s %s", this.name, this.socket.readyState );
-	if ( this.socket.readyState === 1 )
+	log.debug("Waiting for RPC WebSocket client (%s) to be in 'CONNECTED' ready state", this.name );
+
+	if ( this.socket.readyState === 1 ) {
+	    log.silly("WebSocket is already in CONNECTED ready state (%s)", this.socket.readyState );
 	    return;
+	}
 
 	return async_with_timeout(() => {
 	    return new Promise((f,r) => {
-		log.silly("Waiting for open event on %s", this.name );
-
 		const tid		= setTimeout(() => {
+		    log.silly("Checking ready state for RPC WebSocket client (%s): %s", this.name, this.socket.readyState );
 		    if ( this.socket.readyState === 1 )
```

Comment:
Do we still need this check here as we are now returning checking and returning based on this same condition prior to entering async_with_timeout? (lines 17-20 above)

Reply:
Yes, this check happens at a later time so the `readyState` could be different.  This timeout is just a final last check before raising the timeout error